src/JustEat.StatsD/StatsDPublisher.cs

```diff
@@ -47,87 +50,135 @@ public StatsDPublisher(StatsDConfiguration configuration)
+            Span<byte> buffer = stackalloc byte[512];
+            var writer = new FixedBuffer(buffer);
+            _spanFormatter.Decrement(value, bucket, ref writer);
+            Send(writer.Get());
         }
 
         public void Decrement(long value, double sampleRate, string bucket)
         {
-            Send(_formatter.Decrement(value, sampleRate, bucket));
+            Span<byte> buffer = stackalloc byte[512];
+            var writer = new FixedBuffer(buffer);
+            _spanFormatter.Decrement(value, sampleRate, bucket, ref writer);
+            Send(writer.Get());
         }
 
         public void Decrement(long value, double sampleRate, params string[] buckets)
         {
-            Send(_formatter.Decrement(value, sampleRate, buckets));
+            Send(Encoding.UTF8.GetBytes(_formatter.Decrement(value, sampleRate, buckets)));
         }
+
         public void Gauge(double  value, string bucket)
         {
-            Send(_formatter.Gauge(value, bucket));
+            Span<byte> buffer = stackalloc byte[512];
+            var writer = new FixedBuffer(buffer);
+            _spanFormatter.Gauge(value, bucket, ref writer);
+            Send(writer.Get());
         }
+
         public void Gauge(double value, string bucket, DateTime timestamp)
         {
-            Send(_formatter.Gauge(value, bucket, timestamp));
+            Span<byte> buffer = stackalloc byte[512];
+            var writer = new FixedBuffer(buffer);
+            _spanFormatter.Gauge(value, bucket, timestamp, ref writer);
+            Send(writer.Get());
         }
+
         public void Gauge(long value, string bucket)
         {
-            Send(_formatter.Gauge(value, bucket));
+            Span<byte> buffer = stackalloc byte[512];
+            var writer = new FixedBuffer(buffer);
+            _spanFormatter.Gauge(value, bucket, ref writer);
+            Send(writer.Get());
         }
 
         public void Gauge(long value, string bucket, DateTime timestamp)
         {
-            Send(_formatter.Gauge(value, bucket, timestamp));
+            Span<byte> buffer = stackalloc byte[512];
+            var writer = new FixedBuffer(buffer);
+            _spanFormatter.Gauge(value, bucket, timestamp, ref writer);
+            Send(writer.Get());
         }
 
         public void Timing(TimeSpan duration, string bucket)
         {
-            Send(_formatter.Timing(Convert.ToInt64(duration.TotalMilliseconds), bucket));
+            Span<byte> buffer = stackalloc byte[512];
+            var writer = new FixedBuffer(buffer);
+            _spanFormatter.Timing(Convert.ToInt64(duration.TotalMilliseconds), bucket, ref writer);
+            Send(writer.Get());
         }
 
         public void Timing(TimeSpan duration, double sampleRate, string bucket)
         {
-            Send(_formatter.Timing(Convert.ToInt64(duration.TotalMilliseconds), sampleRate, bucket));
+            Span<byte> buffer = stackalloc byte[512];
```

Comment:
Is this a reasonable hard limit, or does it need surfacing in some way?

Reply:
This is just a random constant I choose from the top of my head, we should be able to quickly get the approximate minimum amount of bytes needed based on the metric string length.